data-raw/make_cluster_vcov.R

```diff
@@ -0,0 +1,50 @@
+
+# Create clustered variance-covariance matrix for package tests
+
+library(modmarg)
+library(sandwich)
+library(data.table)
+
+rm(list = ls())
+
+cluster_se <- function(model, cluster, data = model$data){
+  data <- data[complete.cases(data[, names(model$model)]), ]
+
+  # get parameters
+  n <- nrow(data)
+  m <- length(unique(data[, cluster]))
+  k <- length(coef(model))
+
+  # sum over clusters using data.table
+  u <- data.table(estfun(model))
+  u$cluster <- data[, cluster]
+  u_clust <- u[, lapply(.SD, sum), keyby = cluster]
+  # Drop cluster variable, go back to matrix
+  u_clust <- as.matrix(u_clust)[, -1]
+
+  # Sandwich estimator and DFC correction is different for different models
+  if(family(model)[['link']] == 'identity'){
+    dfc <- (m / (m - 1)) * ((n - 1) / (n - k))
+    clust <- dfc * sandwich(model, meat = crossprod(u_clust) / n)
+  } else if(family(model)[['link']] == 'logit'){
+    dfc <- m / (m - 1)
+    clust <- vcov(model) %*% (dfc * t(u_clust) %*% u_clust) %*% vcov(model)
+  }
+  list(clust = clust, stata_dof = m - 1)
+}
+
+# Create sample vcov matrix + degrees of freedom for vcov test
+# Section D.1 here:
```

Comment:
What is D.1? This appears to be ordered by Chapter, then section... But there is no section D in chapter 1. Page number?